### DISCUSSION TITLE
Permanent 502 Bad Gateway

### DISCUSSION THREAD

**Discussion by Lightyear76** [deeztek/Hermes-Secure-Email-Gateway#58] [Support] [answered]
Hello!

After several installation attempts I keep getting an error "502 Bad Gateway nginx/1.18.0 (Ubuntu)".

What am I doing wrong?

Best regards

**Comment by deeztek** [comment]
Hopefully you are not trying to run Hermes in a LXC environment :)

**Reply by Lightyear76** [reply]
Of course not. It's an Ubuntu 20.04 LTS in a virtual machine. However, not container based.

**Reply by deeztek** [reply]
Great. We've had a few issues lately with people trying to run it in LXC environment. Please post the installation log file so that we can take a look.

Thanks

**Reply by Lightyear76** [reply]
[install_log-12-21-2023-0911.log](https://github.com/deeztek/Hermes-Secure-Email-Gateway/files/13783969/install_log-12-21-2023-0911.log)

**Reply by deeztek** [reply]
I see the following issue:

```
Job for hermesseg.service failed because the control process exited with error code.
See "systemctl status hermesseg.service" and "journalctl -xe" for details.
```

Can you post the output of both of the commands above?

**Reply by Lightyear76** [reply]
Hi! Thanks a lot... here are the files... 

[hermesseg.service.txt](https://github.com/deeztek/Hermes-Secure-Email-Gateway/files/13788168/hermesseg.service.txt)
[journalctl-xe.txt](https://github.com/deeztek/Hermes-Secure-Email-Gateway/files/13788169/journalctl-xe.txt)

**Reply by deeztek** [reply]
It's very hard to tell what's happening here. Can you post the contents of the following file?

/var/www/html/server.json

Also, can you start the Hermes SEG server:

systemctl restart hermesseg.service

Then immediately post the output of journalctl -xe

Just making sure you did not use any special characters for any of your passwords.

Thanks

**Reply by Lightyear76** [reply]
[server.json](https://github.com/deeztek/Hermes-Secure-Email-Gateway/files/13791157/server.json)
![Screenshot 2023-12-29 063135](https://github.com/deeztek/Hermes-Secure-Email-Gateway/assets/37901880/12f4dbc7-c826-485c-9c11-3784f789c73e)
[journalctl-xe3.txt](https://github.com/deeztek/Hermes-Secure-Email-Gateway/files/13791165/journalctl-xe3.txt)

**Reply by deeztek** [reply, marked as answer]
okay I found the problem. Ortus changed the commandbox version number structure at some point between 5.8 and 5.9 for some reason. 

So, from a fresh Ubuntu 20.04 server, git clone the Hermes SEG repo and edit the following file:

`Hermes-Secure-Email-Gateway/download/commandbox/server.json `

Change the following parameter in the file:

`"cfengine":"lucee@5.2.9+31"`

to 

`"cfengine":"lucee@5.2.9+031"`

(note the "0" in front of "31"). 

Save it and start the install.

I will be pushing an update to fix this issue soon, but I'm in the middle of an release and it's going to take a few days.

**Reply by Lightyear76** [reply]
Perfect! Now the system works. Thank you very much for this great and fast support!!

**Reply by deeztek** [reply]
np. Thanks for your patience.
